### ISSUE TITLE
Add, remove or change pattern at runtime ?

### ISSUE THREAD

**Issue by dmail** [natefaubion/sparkler#20]
I'm looking for pattern matching solution in order to split the behaviour of a function.
The purpore is that you or a user of your function can add his own behaviour either by changing the pattern triggering a specific function by changing the function behind the pattern.

Let me provide an example illustrating how we could add/change patterns but not how to remove them.

```javascript
function getNumberMeaning {
  42 => 'The meaning of life',
  (*) => 'No meaning'
}

getNumberMeaning(42); //  'The meaning of life',
myGetNumberMeaning = getNumberMeaning.concat(function() {
  42 => 'The meaning of death',
  43 => 'The number after 42'
});

myGetNumberMeaning(42); // 'The meaning of death',
myGetNumberMeaning(43); // 'The number after 42'
````

Can I do that with sparkler ?
If not, do you have any advice on how to achieve this goal ?

**Comment by natefaubion** [comment]
There's an example in the README for extending partial functions with `orElse` https://github.com/natefaubion/sparkler#optional-extensions

Otherwise, no, not with this library, as the patterns are syntactic constructs. You'd want something more like a multi-method dispatch system for that. That's really what https://github.com/natefaubion/matches.js was, but buried underneath a quirky pattern DSL.

**Comment by natefaubion** [comment]
In particular https://github.com/natefaubion/matches.js#patternpatternfn-successfn

**Comment by dmail** [comment]
I missed `orElse` when reading the doc, now I understand its purpose thank you. 
macthes.js is how I discovered sparkler but the thing is: now I've seen sparkler, matches.js lost most of its beauty.

For now using sparkler is a too big step forward in how I'm used to work. Plus I realized I don't absolutely need pattern matching. I can achieve my purpose with something mirroring the behaviour of a switch with the ability to dynamically add/update/remove case statements.

Thank you for your time and your help and the incredible stuff you did with sparkler, I'm sincerely impressed.
